src/components/NewBranchDialog.tsx

```diff
@@ -0,0 +1,440 @@
+          </div>
+        </div>
+        <DialogActions className={actionsWrapperClass}>
+          <input
+            className={classes(buttonClass, cancelButtonClass)}
+            type="button"
+            title="Close this dialog without creating a new branch"
+            value="Cancel"
+            onClick={this._onClose}
+          />
+          <input
+            className={classes(buttonClass, createButtonClass)}
+            type="button"
+            title="Create a new branch"
+            value="Create Branch"
+            onClick={this._onCreate}
+          />
+        </DialogActions>
+      </Dialog>
+    );
+  }
+
+  /**
+   * Renders branch menu items.
+   *
+   * @returns fragment array
+   */
+  private _renderItems() {
+    const current = this.props.model.currentBranch.name;
+    return this.state.branches
+      .slice()
+      .sort(comparator)
+      .map(this._renderItem, this);
+
+    /**
+     * Comparator function for sorting branches.
+     *
+     * @private
+     * @param a - first branch
+     * @param b - second branch
+     * @returns integer indicating sort order
+     */
+    function comparator(a: Git.IBranch, b: Git.IBranch): number {
+      if (a.name === current) {
+        return -1;
+      }
+      return 0;
+    }
+  }
+
+  /**
+   * Renders a branch menu item.
+   *
+   * @param branch - branch
+   * @param idx - item index
+   * @returns fragment
+   */
+  private _renderItem(branch: Git.IBranch, idx: number) {
+    // Perform a "simple" filter... (TODO: consider implementing fuzzy filtering)
+    if (this.state.filter && !branch.name.includes(this.state.filter)) {
+      return null;
+    }
+    const isBase = branch.name === this.state.base;
+    const isCurr = branch.name === this.state.current;
+
+    let isBold;
+    let desc;
+    if (isCurr) {
+      isBold = true;
+      desc = BRANCH_DESC['current'];
+    }
+    return (
+      <ListItem
+        button
+        className={classes(listItemClass, isBase ? activeListItemClass : null)}
+        key={idx}
+        onClick={this._onBranchClickFactory(branch.name)}
+      >
+        <span className={classes(listItemIconClass, 'jp-Icon-16')} />
+        <div className={listItemContentClass}>
+          <p
+            className={classes(
+              listItemTitleClass,
+              isBold ? listItemBoldTitleClass : null
+            )}
+          >
+            {branch.name}
+          </p>
+          {desc ? <p className={listItemDescClass}>{desc}</p> : null}
+        </div>
+      </ListItem>
+    );
+  }
+
+  /**
+   * Adds model listeners.
+   */
+  private _addListeners(): void {
+    // When the repository changes, we're likely to have a new set of branches:
+    this.props.model.repositoryChanged.connect(this._syncState, this);
+
+    // When the HEAD changes, decent probability that we've switched branches:
+    this.props.model.headChanged.connect(this._syncState, this);
```

Comment:
Only this one should appear

Reply:
Removed the `repositoryChanged` listener, but kept the `status` listener, per discussion above.